index.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const differenceWith = require('lodash/differenceWith');
+const isMatch = require('lodash/isMatch');
+const isPlainObject = require('lodash/isPlainObject');
+
+const waitForActions = (store, actions, timeout) => {
+    timeout = timeout || 50;
+
+    if (typeof actions === 'string' || isPlainObject(actions)) {
+        actions = [actions];
+    }
+
+    if (Array.isArray(actions)) {
+        actions = actions.map((value) => typeof value === 'string' ? { type: value } : value);
+    }
+
+    const shouldPromiseResolve = typeof actions === 'function' ?
+        actions :
+        (storeActions) => differenceWith(actions, storeActions, (action, storeAction) => isMatch(storeAction, action)).length === 0;
+
+    // If the store already contains the expected actions, resolve the Promise immediately
+    if (shouldPromiseResolve(store.getActions())) {
+        const promise = Promise.resolve();
+
+        promise.cancel = () => {};
+
+        return promise;
+    }
+
+    let cancel;
+
+    const promise = new Promise((resolve, reject) => {
+        const teardown = () => {
+            clearTimeout(timer);
+            unsubscribe();
+        };
+        const unsubscribe = store.subscribe(() => {
+            if (shouldPromiseResolve(store.getActions())) {
+                teardown();
+                resolve();
+            }
+        });
+        const timer = setTimeout(() => {
```

Comment:
Perhaps makes more sense in terms of logic if we move the `setTimeout` above the `store.subscribe`?

Reply:
`timer` -> `timeoutId`